pkg/http/auth_middleware.go

```diff
@@ -0,0 +1,301 @@
+package http
```

Comment:
Since this is very much S3-specific code, a better package name would actually be `s3` IMHO.

Reply:
ack